Bugsnag/Assets/Bugsnag/Runtime/EndpointConfiguration.cs

```diff
@@ -5,52 +5,85 @@ namespace BugsnagUnity
 {
     public class EndpointConfiguration
     {
-
         private const string DefaultNotifyEndpoint = "https://notify.bugsnag.com";
-
+        private const string HubNotifyEndpoint = "https://notify.insighthub.smartbear.com";
         private const string DefaultSessionEndpoint = "https://sessions.bugsnag.com";
+        private const string HubSessionEndpoint = "https://sessions.insighthub.smartbear.com";
+        private string _customNotifyEndpoint = string.Empty;
+        private string _customSessionEndpoint = string.Empty;
+        internal Uri NotifyEndpoint;
+        internal Uri SessionEndpoint;
+        internal bool IsConfigured = false;
 
-        internal Uri Notify;
-
-        internal Uri Session;
-
-        internal bool NotifyIsCustom => Notify != null && Notify.ToString() != new Uri(DefaultNotifyEndpoint).ToString();
-
-        internal bool SessionIsCustom => Session != null && Session.ToString() != new Uri(DefaultSessionEndpoint).ToString();
-
-        internal bool IsValid
+        internal void Configure(string apiKey)
         {
-            get
+            if (IsConfigured || string.IsNullOrEmpty(apiKey))
             {
-                return Notify != null && Session != null && (NotifyIsCustom && SessionIsCustom || !NotifyIsCustom && !SessionIsCustom);
+                return;
+            }
+            // Check that if one endpoint is customised the other is also customised
+            if (!string.IsNullOrEmpty(_customNotifyEndpoint) && string.IsNullOrEmpty(_customSessionEndpoint))
+            {
+                UnityEngine.Debug.LogWarning("Invalid configuration. endpoints.NotifyEndpoint cannot be set without also setting endpoints.SessionEndpoint. Events will not be sent to Bugsnag.");
+                return;
+            }
+            if (!string.IsNullOrEmpty(_customSessionEndpoint) && string.IsNullOrEmpty(_customNotifyEndpoint))
+            {
+                UnityEngine.Debug.LogWarning("Invalid configuration. endpoints.SessionEndpoint cannot be set without also setting endpoints.NotifyEndpoint. Sessions will not be sent to Bugsnag.");
+                return;
             }
-        }
-
-        internal EndpointConfiguration()
-        {
-            Notify = new Uri(DefaultNotifyEndpoint);
-            Session = new Uri(DefaultSessionEndpoint);
-        }
-
-        public EndpointConfiguration(string notifyEndpoint, string sessionEndpoint)
-        {
             try
             {
-                Notify = new Uri(notifyEndpoint);
+                if (!string.IsNullOrEmpty(_customNotifyEndpoint))
+                {
+                    NotifyEndpoint = new Uri(_customNotifyEndpoint);
+                }
+                else if (apiKey.StartsWith("00000"))
```

Comment:
Suggest extracting to `private const string HubApiPrefix` to help self-document this.

Reply:
I would suggest rather extracting it into its own utility function, something is `isHubApiKey`?